solutions/search/esql-full-text-filter-tutorial.md

```diff
@@ -0,0 +1,411 @@
+
+Sometimes requiring all terms is too strict, but the default OR behavior is too lenient. You can specify a minimum number of terms that must match:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE match(title, "fluffy pancakes breakfast", {"minimum_should_match": 2})
+    | LIMIT 1000
+  """
+}
+```
+
+This query searches the title field to match at least 2 of the 3 terms: "fluffy", "pancakes", or "breakfast".
+
+## Step 4: Search across multiple fields at once
+
+When users enter a search query, they often don't know (or care) whether their search terms appear in a specific field. ES|QL provides ways to search across multiple fields simultaneously:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE title:"vegetarian curry" OR description:"vegetarian curry" OR tags:"vegetarian curry"
+    | LIMIT 1000
+  """
+}
+```
+
+This query searches for "vegetarian curry" across the title, description, and tags fields. Each field is treated with equal importance.
+
+However, in many cases, matches in certain fields (like the title) might be more relevant than others. We can adjust the importance of each field using scoring:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog METADATA _score
+    | WHERE match(title, "vegetarian curry", {"boost": 2.0}) 
+        OR match(description, "vegetarian curry") 
+        OR match(tags, "vegetarian curry")
+    | KEEP title, description, tags, _score
+    | SORT _score DESC
+    | LIMIT 1000
+  """
+}
+```
+
+In this example, we're using the `boost` parameter to make matches in the title field twice as important as matches in other fields. We also request the `_score` metadata field to sort results by relevance.
+
+## Step 5: Filter and find exact matches
+
+Filtering allows you to narrow down your search results based on exact criteria. Unlike full-text searches, filters are binary (yes/no) and do not affect the relevance score. Filters execute faster than queries because excluded results don't need to be scored.
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE category.keyword == "Breakfast"
+    | KEEP title, author, rating, tags
+    | SORT rating DESC
+    | LIMIT 1000
+  """
+}
+```
+
+Note the use of `category.keyword` here. This refers to the [`keyword`](elasticsearch://reference/elasticsearch/mapping-reference/keyword.md) multi-field of the `category` field, ensuring an exact, case-sensitive match.
+
+### Search for posts within a date range
+
+Often users want to find content published within a specific time frame:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE date >= "2023-05-01" AND date <= "2023-05-31"
+    | KEEP title, author, date, rating
+    | LIMIT 1000
+  """
+}
+```
+
+### Find exact matches
+
+Sometimes users want to search for exact terms to eliminate ambiguity in their search results:
```

Comment:
Not sure how we achieve this in ESQL because match uses analysis obs

Reply:
```sql
POST /_query?format=txt
{
  "query": """
    FROM cooking_blog
    | WHERE tags.keyword == "vegetarian"
    | KEEP title, author, rating, tags
    | LIMIT 1000
  """
}
```

Fails with 
`#! Line 3:13: evaluation of [tags.keyword == "vegetarian"] failed, treating result as null. Only first 20 failures recorded.`

Reply:
https://github.com/elastic/docs-content/pull/809/commits/b76bb8b54d87e4fb35ee339bb007ba986e6a0365